.github/workflows/kind_host.yml

```diff
@@ -0,0 +1,137 @@
+        type: string
+        default: 'false'
+
+env:
+  RELEASE_BRANCH: ${{ github.event.pull_request.base.ref || github.ref_name }}
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  tests:
+    name: Hosted mode Tests # Part of the check name, be careful when changing.
+
+    runs-on: ubuntu-latest
+    env:
+      REGISTRY: localhost:5000
+    strategy:
+      # Don't skip all tests if one fails - others may still have useful info
+      fail-fast: false
+
+    steps:
+    - name: Checkout Component Repository
+      # Checkout a specific component when called from another repository
+      if: ${{ github.event.repository.name != 'governance-policy-framework' }}
+      uses: actions/checkout@v3
+      with:
+        # `repository` is inferred as the "caller" repository 
+        path: component
+        # `ref` is inferred as the new commit in the "caller" repository
+
+    - name: Checkout Policy Framework
+      # Checkout a stable branch when called from another repository
+      if: ${{ github.event.repository.name != 'governance-policy-framework' }}
+      uses: actions/checkout@v3
+      with:
+        repository: stolostron/governance-policy-framework
+        path: framework
+        ref: ${{ env.RELEASE_BRANCH }} # like main or release-*
+
+    - name: Checkout Policy Framework
+      # Checkout like "usual" when called from this repository
+      if: ${{ github.event.repository.name == 'governance-policy-framework' }}
+      uses: actions/checkout@v3
+      with:
+        # `repository` is inferred as the "caller" repository 
+        path: framework
+        # `ref` is inferred as the new commit
+
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version-file: framework/go.mod
+
+    - name: Bootstrap the KinD Cluster for hosted mode
+      working-directory: framework
+      run: |
+        echo "::group::make e2e-dependencies"
+        make e2e-dependencies
+        echo "::endgroup::"
+
+        echo "::group::make setup hosted mode"
+        make kind-bootstrap-hosted
+        echo "::endgroup::"
+        echo "::group::Saving kubeconfig paths for use in other steps"
+    
+        echo "MANAGED_KUBECONFIG=$(pwd)/kubeconfig_managed" >> $GITHUB_ENV
+        echo "HUB_KUBECONFIG=$(pwd)/kubeconfig_hub" >> $GITHUB_ENV
+        echo "HUB_INTERNAL_KUBECONFIG=$(pwd)/kubeconfig_hub_internal" >> $GITHUB_ENV
+        echo "::endgroup::"
+    
+        echo "::group::test for patch step"
+        export MANAGED_CONFIG=$(pwd)/kubeconfig_managed
+        export HUB_CONFIG=$(pwd)/kubeconfig_hub
+        export HUB_CONFIG_INTERNAL=$(pwd)/kubeconfig_hub_internal
+        echo "::endgroup::"
+
+    - name: Patch Component Image
+      if: ${{ github.event.repository.name != 'governance-policy-framework' }}
+      working-directory: component
+      env:
+        HUB_ONLY: ${{ inputs.hub_only_component }}
+        WATCH_NAMESPACE: managed
+      run: |
+        if [[ "${HUB_ONLY}" == "true" ]]; then
```

Comment:
This if statement can be removed since we aren't doing "HUB_ONLY" in the hosted mode tests.

Reply:
@mprahl Isn't this referring to whether the component is hub only, like the propagator? https://github.com/stolostron/governance-policy-propagator/blob/main/.github/workflows/framework-kind.yaml